runtime/src/comm/gasnet/comm-gasnet.c

```diff
@@ -814,7 +893,58 @@ void  chpl_comm_put(void* addr, c_nodeid_t node, void* raddr,
       chpl_comm_commDiagnostics.put++;
       chpl_sync_unlock(&chpl_comm_diagnostics_sync);
     }
-    gasnet_put(node, raddr, addr, size); // node, dest, src, size
+
+    // Handle remote address not in remote segment.
+#ifdef GASNET_SEGMENT_EVERYTHING
```

Comment:
The code would be cleaner if the #ifdef here were moved inside of chpl_comm_is_in_segment(). Currently that fn actually has such an #ifdef in it, with unfortunately exactly the opposite sense. Is there no way to reconcile these two? The combination of the behavior of the #ifdef outside chpl_comm_is_in_segment() and the identical but opposite-sensed one inside it seems bizarre.